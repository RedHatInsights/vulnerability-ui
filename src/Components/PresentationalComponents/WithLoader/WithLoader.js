import propTypes from 'prop-types';
import React from 'react';
import ContentLoader, { BulletList, List } from 'react-content-loader';

const WithLoader = props => {
    if (props === undefined || props.loading !== false) {
        switch (props.variant) {
            case 'cvePageOverviewItem':
                return <CVEPageOverviewItemLoader />;
            case 'list':
                return <List />;
            case 'bulletList':
                return <BulletList />;
            case 'trendBox':
                return <TrendBoxLoader />;
            case 'currentVulnerabilitiesItem':
                return <CurrentVulnerabilitiesItemLoader />;
            default:
                return <ContentLoader style={{ maxHeight: '100px', width: '100%' }} width="100%" />;
        }
    }

    return props.children;
};

// Loaders
const CVEPageOverviewItemLoader = () => {
    return (
        <ContentLoader height={61} width={120}>
            <rect x="35" y="0" rx="0" ry="0" width="50" height="35" />
            <rect x="0" y="45" rx="0" ry="0" width="120" height="16" />
        </ContentLoader>
    );
};

const TrendBoxLoader = () => {
    return (
        <ContentLoader height={100} width={300}>
            <rect x="388.58" y="109.13" rx="0" ry="0" width="0" height="0" />
            <rect x="11.58" y="16.13" rx="0" ry="0" width="37" height="62" />
            <rect x="52.58" y="52.13" rx="0" ry="0" width="17" height="26" />
            <rect x="76.58" y="52.13" rx="0" ry="0" width="17" height="26" />
            <rect x="4.58" y="82.13" rx="0" ry="0" width="154" height="22" />
        </ContentLoader>
    );
};

const CurrentVulnerabilitiesItemLoader = () => {
    return (
        <ContentLoader height={50} width={200} speed={2} primaryColor="#f3f3f3" secondaryColor="#ecebeb">
            <rect x="82.58" y="15.13" rx="0" ry="0" width="110.88" height="21.42" />
            <circle cx="25" cy="25" r="25" />
        </ContentLoader>
    );
};

WithLoader.propTypes = {
    loading: propTypes.bool,
    variant: propTypes.string
};

export default WithLoader;
