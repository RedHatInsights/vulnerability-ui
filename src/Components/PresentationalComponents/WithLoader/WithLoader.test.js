import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import WithLoader from './WithLoader';

describe('WithLoader component', () => {
    it('should render without props', () => {
        const wrapper = shallow(<WithLoader />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render with cvePageOverviewItem variant', () => {
        const wrapper = shallow(<WithLoader variant={'cvePageOverviewItem'} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render with list variant', () => {
        const wrapper = shallow(<WithLoader variant={'list'} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render with bulletList variant', () => {
        const wrapper = shallow(<WithLoader variant={'bulletList'} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render with trendBox variant', () => {
        const wrapper = shallow(<WithLoader variant={'trendBox'} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render with currentVulnerabilitiesItem variant', () => {
        const wrapper = shallow(<WithLoader variant={'currentVulnerabilitiesItem'} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render with loading prop is undefined', () => {
        const wrapper = shallow(<WithLoader loading={undefined} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render with loading prop is true', () => {
        const wrapper = shallow(<WithLoader loading={true} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render with loading prop is false', () => {
        const wrapper = shallow(<WithLoader loading={false} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
