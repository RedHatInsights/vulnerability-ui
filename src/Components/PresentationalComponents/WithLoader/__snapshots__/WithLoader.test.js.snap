// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithLoader component should render with bulletList variant 1`] = `<BulletListStyle />`;

exports[`WithLoader component should render with currentVulnerabilitiesItem variant 1`] = `<CurrentVulnerabilitiesItemLoader />`;

exports[`WithLoader component should render with cvePageOverviewItem variant 1`] = `<CVEPageOverviewItemLoader />`;

exports[`WithLoader component should render with list variant 1`] = `<ListStyle />`;

exports[`WithLoader component should render with loading prop is false 1`] = `""`;

exports[`WithLoader component should render with loading prop is true 1`] = `
<Holder
  style={
    Object {
      "maxHeight": "100px",
      "width": "100%",
    }
  }
  width="100%"
/>
`;

exports[`WithLoader component should render with loading prop is undefined 1`] = `
<Holder
  style={
    Object {
      "maxHeight": "100px",
      "width": "100%",
    }
  }
  width="100%"
/>
`;

exports[`WithLoader component should render with trendBox variant 1`] = `<TrendBoxLoader />`;

exports[`WithLoader component should render without props 1`] = `
<Holder
  style={
    Object {
      "maxHeight": "100px",
      "width": "100%",
    }
  }
  width="100%"
/>
`;
